src/sys/event.rs

```diff
@@ -37,43 +97,66 @@ libc_enum! {
     #[cfg_attr(target_os = "netbsd", repr(u32))]
     #[cfg_attr(not(target_os = "netbsd"), repr(i16))]
     #[non_exhaustive]
+    /// Kqueue filter types.  These are all the different types of event that a
+    /// kqueue can notify for.
     pub enum EventFilter {
+        /// Notifies on the completion of a POSIX AIO operation.
         EVFILT_AIO,
-        /// Returns whenever there is no remaining data in the write buffer
         #[cfg(target_os = "freebsd")]
+        /// Returns whenever there is no remaining data in the write buffer
         EVFILT_EMPTY,
         #[cfg(target_os = "dragonfly")]
+        /// Takes a descriptor as the identifier, and returns whenever one of
+        /// the specified exceptional conditions has occurred on the descriptor.
         EVFILT_EXCEPT,
         #[cfg(any(target_os = "dragonfly",
                   target_os = "freebsd",
                   target_os = "ios",
                   target_os = "macos"))]
+        /// Establishes a file system monitor.
         EVFILT_FS,
         #[cfg(target_os = "freebsd")]
+        /// Notify for completion of a list of POSIX AIO operations.
+        /// # See Also
+        /// [lio_listio(2)](https://www.freebsd.org/cgi/man.cgi?query=lio_listio)
         EVFILT_LIO,
         #[cfg(any(target_os = "ios", target_os = "macos"))]
+        /// Mach portsets
         EVFILT_MACHPORT,
```

Comment:
This part may be unsound; I'm not sure about mach port support here but it's definitely not just an FD.

Reply:
What might not just be an FD?  And what part might be unsound?  This just defines the filter constant.